### PR TITLE
Feat/5285 electrum optimization

### LINKED ISSUE #5267: your wallet uses less efficient method of querying electrs, consider contacting the developer of your wallet
Using Trezor Suite [22.3.2](https://github.com/trezor/trezor-suite/releases/tag/v22.3.2) and electrs v0.9.6 as a custom Bitcoin backend I observe a lot of the following messages in the electrs log for `get_balance`, `listunspent` and `get_history` queries:

```
INFO  electrs::electrum] your wallet uses less efficient method of querying electrs, 
consider contacting the developer of your wallet. 
Reason: blockchain.scripthash.get_balance called for unsubscribed scripthash: <REDACTED SCRIPTHASH>
```

```
NFO  electrs::electrum] your wallet uses less efficient method of querying electrs, 
consider contacting the developer of your wallet. 
Reason: blockchain.scripthash.listunspent called for unsubscribed scripthash: <REDACTED SCRIPTHASH>
```

```
INFO  electrs::electrum] your wallet uses less efficient method of querying electrs, 
consider contacting the developer of your wallet. 
Reason: blockchain.scripthash.get_history called for unsubscribed scripthash: <REDACTED SCRIPTHASH>
```

While Trezor Suite works with electrs, experience is very slow. I would expect some caching on recurring usage of the same Trezor device (wallet addresses) with electrs and I wonder if subscribing to the scripthashes first would improve the experience significantly? See a related comment: https://github.com/romanz/electrs/issues/608#issuecomment-960497420

### LINKED ISSUE #5285: Electrum server backend: each scripthash is called twice
According to https://github.com/romanz/electrs/issues/702 the Suite queries each scripthash twice.

Is that intentional or is this a bug?

### PR DESCRIPTION
**Pull request by marekrjpolak**
Fixes #5267,  fixes #5285 and other minor bugs.

- https://github.com/trezor/trezor-suite/commit/a3f8268a5db6080ef7fee557d89f3671b8ad3212 - subscribes to every scripthash before any `get_history`, `get_balance` or `listunspent` request and therefore allows more effective caching. When tested, it shortened the discovery time by 40-60 % without Tor and 10-15 % with Umbrel over Tor.
- https://github.com/trezor/trezor-suite/commit/f975eb32365c41817a514641e26579a736906329 - now it should be possible to subscribe to testnet/regtest addresses (before that, only mainnet subscriptions worked).
- https://github.com/trezor/trezor-suite/commit/e1caacdf83df5dae5bfd52c82b0646acabc322c3 - fixes last block caching.
- https://github.com/trezor/trezor-suite/commit/a4f4c06d7b8aa11a23cb5a053e0e658a8590af10 - throttles transaction notifications so there isn't a separate notification for every changed scripthash under the same account at the same time.
- https://github.com/trezor/trezor-suite/commit/60fb178a2581238d893574ca788afe3bcdae11ce - sorts the transaction history returned by `getAccountInfo` from the latest transaction to the oldest one and therefore fixes the bug with occasional transaction disappearing when using Electrum.

### What to test ###
All these changes are related to the custom Electrum backend only (for BTC, TEST and REGTEST), so it's the only thing which must be tested (discovery, sending, receiving, notifications, ...). The only exception is the throttling mechanism in https://github.com/trezor/trezor-suite/commit/a4f4c06d7b8aa11a23cb5a053e0e658a8590af10 which affects all the backend types (but in theory shouldn't change anything). So please make sure that transactions notification works as expected even on different backend types.

### CODE REVIEW

File: packages/blockchain-link/src/workers/throttler.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        }
+        this.timeouts[id] = setTimeout(() => {
+            callback();
+            clearTimeout(this.timeouts[id]);
+            delete this.timeouts[id];
+        }, this.delay);
+    }
+
+    dispose() {
+        Object.values(this.timeouts).map(clearTimeout);
```
Comment (szymonlesisz) [inline packages/blockchain-link/src/workers/throttler.ts]: delete all values in `this.timeouts` maybe?

File: packages/blockchain-link/src/types/electrum.ts [hunk 2]
```diff
@@ -59,7 +59,7 @@ export type TransactionVerbose = {
     vsize: number;
     weight: number;
     locktime: number;
-    confirmations: number;
+    confirmations: number | undefined;
```
Comment (szymonlesisz) [inline packages/blockchain-link/src/types/electrum.ts]: `confirmations?: number`

**Review by szymonlesisz — COMMENTED** [review]
lgtm :) found two nitpicks but it's up to you if you solve them, it shoudn't have influence on the runtime

File: packages/blockchain-link/src/types/electrum.ts (hunk 2 above)
Comment (marekrjpolak) [reply packages/blockchain-link/src/types/electrum.ts]: https://github.com/trezor/trezor-suite/pull/5454/commits/8260da33743acb0edee4ead8c0fe14ef43db9198

File: packages/blockchain-link/src/workers/throttler.ts (hunk 1 above)
Comment (marekrjpolak) [reply packages/blockchain-link/src/workers/throttler.ts]: https://github.com/trezor/trezor-suite/pull/5454/commits/9987e3fc0304c40b2493acd5319f0f81643325f2 and I've added unit test to make sure `Throttler` works as expected.

**Review by szymonlesisz — APPROVED** [review]

**Pull request merged**